cmd/bbolt/page_command.go

```diff
@@ -1,105 +1,97 @@
 package main
 
 import (
-	"bytes"
-	"flag"
 	"fmt"
-	"io"
 	"os"
-	"strings"
 
+	"github.com/spf13/cobra"
 	"go.etcd.io/bbolt/internal/common"
 	"go.etcd.io/bbolt/internal/guts_cli"
 )
 
-// pageCommand represents the "page" command execution.
-type pageCommand struct {
-	baseCommand
-}
+// NewPageCommand creates a new Cobra command for the "page" subcommand.
+func NewPageCommand() *cobra.Command {
+	var all bool
+	var formatValue string
+
+	// Create a new Cobra command for "page".
+	cmd := &cobra.Command{
+		Use:   "page PATH pageid [pageid...]",
+		Short: "Print page(s) from a Bolt DB",
+		Long:  "Print one or more pages from the specified Bolt DB.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Validate that PATH is provided.
+			if len(args) < 1 {
+				return fmt.Errorf("path to the database is required")
+			}
 
-// newPageCommand returns a pageCommand.
-func newPageCommand(m *Main) *pageCommand {
-	c := &pageCommand{}
-	c.baseCommand = m.baseCommand
-	return c
-}
+			path := args[0]
+			if _, err := os.Stat(path); os.IsNotExist(err) {
+				return fmt.Errorf("file not found: %s", path)
+			}
 
-// Run executes the command.
-func (cmd *pageCommand) Run(args ...string) error {
-	// Parse flags.
-	fs := flag.NewFlagSet("", flag.ContinueOnError)
-	help := fs.Bool("h", false, "")
-	all := fs.Bool("all", false, "list all pages")
-	formatValue := fs.String("format-value", "auto", "One of: "+FORMAT_MODES+" . Applies to values on the leaf page.")
-
-	if err := fs.Parse(args); err != nil {
-		return err
-	} else if *help {
-		fmt.Fprintln(cmd.Stderr, cmd.Usage())
-		return ErrUsage
-	}
+			// Handle the case when `--all` flag is passed.
+			if all {
+				return printAllPages(path, formatValue)
+			}
 
-	// Require database path and page id.
-	path := fs.Arg(0)
-	if path == "" {
-		return ErrPathRequired
-	} else if _, err := os.Stat(path); os.IsNotExist(err) {
-		return ErrFileNotFound
-	}
+			// Get page IDs from the arguments.
```

Comment:
so it is better to use `pageFunc()` and inline each arg by name. 

then in 

```
 		RunE: func(cmd *cobra.Command, args []string) error {
```

above you can extract the args as you wish 

cc @ahrtr @fuweid @ivanvc please correct me if i am wrong